Only in fckeditor.patched/: fck__drupalbreak.gif
diff -r -u fckeditor/fckeditor.config.js fckeditor.patched/fckeditor.config.js
--- fckeditor/fckeditor.config.js	2008-12-12 19:45:25.000000000 +0000
+++ fckeditor.patched/fckeditor.config.js	2009-01-15 12:46:15.000000000 +0000
@@ -55,7 +55,7 @@
 ] ;
 
 FCKConfig.ToolbarSets["DrupalBasic"] = [
-['FontFormat','-','Bold','Italic','-','OrderedList','UnorderedList','-','Link','Unlink', 'Image']
+['Bold','Italic','-','RemoveFormat','Undo','Redo','-','OrderedList','UnorderedList','-','Link','Unlink', 'Image','DrupalBreak']
 ] ;
 
 //This toolbar should work fine with "Filtered HTML" filter
@@ -120,4 +120,4 @@
 // [ Left, Center, Right, Justified ]
 FCKConfig.JustifyClasses = ['rteleft','rtecenter','rteright','rtejustify'] ;
 //Set to 'encode' if you want to obfuscate emails with javascript
-FCKConfig.EMailProtection = 'none' ;
\ No newline at end of file
+FCKConfig.EMailProtection = 'none' ;
diff -r -u fckeditor/fckeditor.css fckeditor.patched/fckeditor.css
--- fckeditor/fckeditor.css	2008-09-16 10:44:46.000000000 +0100
+++ fckeditor.patched/fckeditor.css	2009-01-15 12:13:07.000000000 +0000
@@ -28,4 +28,18 @@
   background-image: url(plugins/imgassist/imgassist.gif);
   background-position: 50% 50%;
   border: 1px dashed #dddddd;
-}
\ No newline at end of file
+}
+.FCK__DrupalBreak {
+	background-position: center center;
+	background-image: url(fck__drupalbreak.gif);
+	background-repeat: no-repeat;
+	clear: both;
+	display: block;
+	float: none;
+	width: 100%;
+	border-top: #999999 1px	dotted;
+	border-bottom: #999999 1px dotted;
+	border-right: 0px;
+	border-left: 0px;
+	height:	13px;
+}
diff -r -u fckeditor/fckeditor.utils.js fckeditor.patched/fckeditor.utils.js
--- fckeditor/fckeditor.utils.js	2008-12-12 20:27:03.000000000 +0000
+++ fckeditor.patched/fckeditor.utils.js	2009-01-15 12:42:29.000000000 +0000
@@ -238,6 +238,10 @@
 
   //Img_Assist integration
   IntegrateWithImgAssist();
+
+  // Upload module image insert into fckeditor wants this
+  window.oooFckName = editorInstance.Name;
+  window.oooFCKeditorAPI = FCKeditorAPI;
 }
 
 function FCKeditorReplaceTextarea(textarea_id, oFCKeditor, xss_check)
Only in fckeditor.patched/: .fckeditor.utils.js.swp
diff -r -u fckeditor/plugins/drupalbreak/fckplugin.js fckeditor.patched/plugins/drupalbreak/fckplugin.js
--- fckeditor/plugins/drupalbreak/fckplugin.js	2008-11-19 12:00:07.000000000 +0000
+++ fckeditor.patched/plugins/drupalbreak/fckplugin.js	2009-01-15 12:16:18.000000000 +0000
@@ -44,7 +44,7 @@
 		case 'Break' :
 			this.RemoveOldBreaks();
 			var e = FCK.EditorDocument.createComment( 'break' ) ;
-			oFakeImage = FCK.InsertElement( FCKDocumentProcessor_CreateFakeImage( 'FCK__PageBreak', e ) ) ;
+			oFakeImage = FCK.InsertElement( FCKDocumentProcessor_CreateFakeImage( 'FCK__DrupalBreak', e ) ) ;
 			oFakeImage.setAttribute( "_drupalbreak", "true" ) ;
 			oFakeImage.style.borderTop = oFakeImage.style.borderBottom = pBreakBorderStyle ;
 			this.MoveBreakOutsideElement();
@@ -166,7 +166,7 @@
 				if (re.test(node.nodeValue))
 					PContent = FCKConfig.ProtectedSource.Revert('<!--' + node.nodeValue + '-->', false);
 				if (node.nodeValue == 'break' || PContent == '<!--break-->') {
-					var oFakeImage = FCKDocumentProcessor_CreateFakeImage( 'FCK__PageBreak', node.cloneNode(true) ) ;
+					var oFakeImage = FCKDocumentProcessor_CreateFakeImage( 'FCK__DrupalBreak', node.cloneNode(true) ) ;
 					oFakeImage.setAttribute( "_drupalbreak", "true" ) ;
 					oFakeImage.style.borderTop = oFakeImage.style.borderBottom = pBreakBorderStyle ;
 					node.parentNode.insertBefore( oFakeImage, node ) ;
@@ -195,4 +195,4 @@
 		
 	result = result.replace( re, '$2' );
 	return result ;
-}
\ No newline at end of file
+}
diff -r -u fckeditor/plugins/drupalbreak/lang/en.js fckeditor.patched/plugins/drupalbreak/lang/en.js
--- fckeditor/plugins/drupalbreak/lang/en.js	2008-11-19 12:00:08.000000000 +0000
+++ fckeditor.patched/plugins/drupalbreak/lang/en.js	2009-01-15 12:16:29.000000000 +0000
@@ -1,2 +1,2 @@
 FCKLang.DrupalBreakTooltip = 'Insert Teaser Break' ;
-FCKLang.DrupalBreakTitle = 'Teaser' ;
+FCKLang.DrupalBreakTitle = 'Summary / body separator' ;
